packages/container/lib/container.js

```diff
@@ -330,8 +330,7 @@ function lookup(container, fullName, options = {}) {
 }
 
 function isSingletonClass(container, fullName, { instantiate, singleton }) {
-  return (singleton !== false && isSingleton(container, fullName)) &&
-         (instantiate === false && !shouldInstantiate(container, fullName));
+  return singleton !== false && isSingleton(container, fullName) && !instantiate && !shouldInstantiate(container, fullName);
 }
 
 function isSingletonInstance(container, fullName, { instantiate, singleton }) {
```

Comment:
I was looking at this [pull request](https://github.com/emberjs/ember.js/pull/15014/files). Should I replace it?